dkg/runner.go

```diff
@@ -69,78 +76,79 @@ func (r *Runner) ProcessMsg(msg *SignedMessage) (bool, map[types.OperatorID]*Sig
-			return false, nil, errors.Wrap(err, "PartialDepositData invalid")
+			return false, errors.Wrap(err, "PartialDepositData invalid")
 		}
 
 		if found := r.DepositDataSignatures[msg.Signer]; found == nil {
 			r.DepositDataSignatures[msg.Signer] = depSig
 		} else if !bytes.Equal(found.Signature, msg.Signature) {
-			return false, nil, errors.New("inconsistent partial signature received")
+			return false, errors.New("inconsistent partial signature received")
 		}
 
 		if len(r.DepositDataSignatures) == int(r.InitMsg.Threshold) {
 			// reconstruct deposit data sig
 			depositSig, err := r.reconstructDepositDataSignature()
 			if err != nil {
-				return false, nil, errors.Wrap(err, "could not reconstruct deposit data sig")
+				return false, errors.Wrap(err, "could not reconstruct deposit data sig")
 			}
 
 			// encrypt Operator's share
-			encryptedShare, err := r.config.Signer.Encrypt(r.Operator.EncryptionPubKey, r.KeyGenOutput.Share.Serialize())
+			encryptedShare, err := r.config.Signer.Encrypt(r.Operator.EncryptionPubKey, r.KeygenOutcome.KeyGenOutput.Share.Serialize())
 			if err != nil {
-				return false, nil, errors.Wrap(err, "could not encrypt share")
+				return false, errors.Wrap(err, "could not encrypt share")
 			}
 
 			ret, err := r.generateSignedOutput(&Output{
 				RequestID:            r.Identifier,
 				EncryptedShare:       encryptedShare,
-				SharePubKey:          r.KeyGenOutput.Share.GetPublicKey().Serialize(),
-				ValidatorPubKey:      r.KeyGenOutput.ValidatorPK,
+				SharePubKey:          r.KeygenOutcome.KeyGenOutput.Share.GetPublicKey().Serialize(),
+				ValidatorPubKey:      r.KeygenOutcome.KeyGenOutput.ValidatorPK,
 				DepositDataSignature: depositSig,
 			})
 			if err != nil {
-				return false, nil, errors.Wrap(err, "could not generate dkg SignedOutput")
+				return false, errors.Wrap(err, "could not generate dkg SignedOutput")
 			}
 
 			r.OutputMsgs[r.Operator.OperatorID] = ret
 			if err := r.signAndBroadcastMsg(ret, OutputMsgType); err != nil {
-				return false, nil, errors.Wrap(err, "could not broadcast SignedOutput")
+				return false, errors.Wrap(err, "could not broadcast SignedOutput")
 			}
-			return false, nil, nil
+			return false, nil
 		}
 	case OutputMsgType:
 		output := &SignedOutput{}
 		if err := output.Decode(msg.Message.Data); err != nil {
-			return false, nil, errors.Wrap(err, "could not decode SignedOutput")
+			return false, errors.Wrap(err, "could not decode SignedOutput")
 		}
 
 		if err := r.validateSignedOutput(output); err != nil {
-			return false, nil, errors.Wrap(err, "signed output invali")
+			return false, errors.Wrap(err, "signed output invali")
 		}
 
 		r.OutputMsgs[msg.Signer] = output
 		// GLNOTE: Actually we need every operator to sign instead only the quorum!
 		if len(r.OutputMsgs) == len(r.InitMsg.OperatorIDs) {
-			return true, r.OutputMsgs, nil
+			err := r.config.Network.StreamDKGOutput(r.OutputMsgs)
```

Comment:
why is it moved from node to here?

Reply:
We need to stream both Output and Blame. To make it consistent, we either do both in node or in runner. To stream it in node, we need to return both blame and keygen output to node. So I think it's cleaner if we stream both in runner.